src/services/codefixes/codeFixProvider.ts

```diff
@@ -0,0 +1,53 @@
+/* @internal */
+namespace ts {
+    export interface CodeFix {
+        name: string;
+        errorCodes: string[];
+        getCodeActions(context: CodeFixContext): CodeAction[];
+    }
+
+    export interface CodeFixContext {
+        errorCode: string;
+        sourceFile: SourceFile;
+        span: TextSpan;
+        checker: TypeChecker;
+        newLineCharacter: string;
+    }
+
+    export namespace codeFix {

```

Comment:
Style nit: It feels odd to have a lower-camelcase namespace. I know we already have `jsTypings`, but that feels odd, too. Given that we favor lowercase styled namespaces `codefix` seems more appropriate.